### PR TITLE
PLNSRVCE-1281: Update sprayproxy component for staging kustomize overlay

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
### Update staging sprayproxy component.

- Update sprayproxy deployment link and image, but only for staging kustomize overlay.
-- Deployment should be updated to include harden sprayproxy api route fix.
-- Image should be updated to include improvement: log forwarding requests.
- Moved base spraypoxy url to the overlay kustomization.yaml to provide opportunity apply deployment changes only for staging vs. production.

### CODE REVIEW

**Comment by avinal** [comment]
Please update PR description.

**Comment by adambkaplan** [comment]
Agree with the other reviewers - this pull request should have a fuller description. I would go further and say the commit message should also have similar - if not longer - level of detail.

Tekton has excellent references/guidance on how to create well written commit message: https://github.com/tektoncd/community/blob/main/standards.md#commit-messages

They also have good guidance on how to write a good pull request (some things, like TEPs, are specific to Tekton): https://github.com/tektoncd/community/blob/main/standards.md#pull-request-description

File: components/sprayproxy/production/kustomization.yaml [hunk 1]
```diff
@@ -2,6 +2,12 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 resources:
   - ../base
+
+images:
+  - name: ko://github.com/redhat-appstudio/sprayproxy
+    newName: quay.io/redhat-appstudio/sprayproxy
+    newTag: b488cf7dada7cdf4767fd32f041c6ea3507f0043
```
[the remark this answers is not in the document]
Comment (AndrienkoAleksandr) [reply components/sprayproxy/production/kustomization.yaml:9]: Pr description updated.

**Comment by AndrienkoAleksandr** [comment]
> Agree with the other reviewers - this pull request should have a fuller description. I would go further and say the commit message should also have similar - if not longer - level of detail.

I updated pull request description and commit message.

Also I moved base spraypoxy url to the overlay kustomization.yaml to provide opportunity apply deployment changes only for staging.

**Comment by gabemontero** [comment]
the e2e failure looks unrelated:  `timed out when waiting for component component-python-flask to be ready in e2e-demos-sucm-tenant namespace`

I'll post a message in the qe forum

**Comment by gabemontero** [comment]
same known e2e HAS/git component create flake

see https://redhat-internal.slack.com/archives/C02FANRBZQD/p1681834465373529 for the tl;dr

/retest

**Comment by gabemontero** [comment]
I'm going to add some more e2e-tests debug as a result of https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/redhat-appstudio_infra-deployments/1666/pull-ci-redhat-appstudio-infra-deployments-main-appstudio-e2e-tests/1648678676826427392

**Comment by gabemontero** [comment]
gitops couldn't get off the ground in last run, though there were a ton openshift hypershift issues in CI yesterday

/retest

**Pull request merged**